src/filter.rs

```diff
@@ -76,3 +76,181 @@ impl Filter {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    // TODO Brian: Find a unique access point for the following fn's, repeated in messages.rs.
+    fn generate_random_u8() -> u8 {
```

Comment:
then don't have this function :)

(I refered to line 51 of types.rs of current master for another way of getting the random u8, but best to stick with this paradigm as it's most used throughout the code)